### PR TITLE
feat(esl-traversing-query): add ability to restrict global scope

### PR DESCRIPTION
**Pull request by Anna-MariiaPetryk**
Closes: #522

### CODE REVIEW

File: src/modules/esl-traversing-query/core/esl-traversing-query.ts [hunk 1]
```diff
@@ -94,21 +94,22 @@ export class TraversingQuery {
     return uniq(result);
   }
 
-  static traverse(query: string, findFirst: boolean, base?: Element) {
+  static traverse(query: string, findFirst: boolean, base?: Element | null, scope: Element | null | Document = document) {
+    if (scope === null) throw new Error();
```
Comment (ala-n) [inline src/modules/esl-traversing-query/core/esl-traversing-query.ts:98]: In case we throw an error - there is no reason to allow null value at all

Comment (ala-n) [reply src/modules/esl-traversing-query/core/esl-traversing-query.ts:98]: I'd recommend restricting type and getting rid of an error (the user will have it anyway if he passed a null)
From an API point of view, we are also do not allow undefined or null scope, it is the required field for TraversingQuery
`base` element accepts null just for the purpose of being searched using query without base, e.g. queries like `#id::next` and etc.

**Review by ala-n — CHANGES_REQUESTED** [review]

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

**Pull request merged**